keylime/cloud_verifier_tornado.py

```diff
@@ -573,6 +573,8 @@ def post(self):
                         logger.info("Allowlist data not provided with request! Using default empty allowlist.")
                         ima_policy = json.dumps(ima.process_ima_policy(ima.EMPTY_ALLOWLIST, []))
 
+                    # Although legit in theory, the following initialization breaks the CI tests
+                    # ima_policy_stored = None
```

Comment:
I don't see that this one has to be considered to be initialized since it's only used under the `if ima_policy`. I would drop this patch.

Reply:
Unfortunately, that is not true, in fact the type checker is (obviously) complaining about the variable being possibly unbound: 

https://github.com/mdrocco/keylime/blob/chore-fix-type-issues-ima/keylime/cloud_verifier_tornado.py#L610

Reply:
I think this really highlights some hidden problem(s) in the code, as that initialization looks sound to me. Should I open a separate issue to unclog this PR or we want to figure this out before going on?

Reply:
From my PoV, go ahead and file a separate issue.

Reply:
Why is this initialization before the if clause necessary? I only see it being used further under the `if`:
```

                        try:
                            ima_policy_stored = (
                                session.query(VerifierAllowlist).filter_by(name=ima_policy_name).one_or_none()
                            )
                        except SQLAlchemyError as e:
                            logger.error(
                                "SQLAlchemy Error while retrieving stored ima policy for agent ID %s: %s", agent_id, e
                            )
                            raise
                        try:
                            if ima_policy_stored is None:
                                ima_policy_stored = VerifierAllowlist(**ima_policy_db_format)
                                session.add(ima_policy_stored)
                                session.commit()
                        except SQLAlchemyError as e:
                            logger.error("SQLAlchemy Error while updating ima policy for agent ID %s: %s", agent_id, e)
                            raise
```

Reply:
as I pointed out above, it is also referenced outside the `if`: https://github.com/keylime/keylime/blob/8e417f5b99f0dea0f9f36e26ab15549fcd3e1844/keylime/cloud_verifier_tornado.py#L608

Reply:
right ...

Reply:
I updated the existing issue, please take a look: https://github.com/keylime/keylime/issues/1143#issuecomment-1318795728